demo/style_configurator.c

```diff
@@ -0,0 +1,1048 @@
+
+// TODO design decisions
+// plural or not?  ie style_button or style_buttons?
+// use the duplicate array method, or just let the user
+// manually set those after calling the function by accessing ctx->style->*?
+
+static const char* symbols[NK_SYMBOL_MAX] =
+{
+    "NONE",
+    "X",
+    "UNDERSCORE",
+    "CIRCLE_SOLID",
+    "CIRCLE_OUTLINE",
+    "RECT_SOLID",
+    "RECT_OUTLINE",
+    "TRIANGLE_UP",
+    "TRIANGLE_DOWN",
+    "TRIANGLE_LEFT",
+    "TRIANGLE_RIGHT",
+    "PLUS",
+    "MINUS"
+};
+
+
+
+static void
+style_rgb(struct nk_context* ctx, const char* name, struct nk_color* color)
+{
+	nk_label(ctx, name, NK_TEXT_LEFT);
+	if (nk_combo_begin_color(ctx, *color, nk_vec2(nk_widget_width(ctx), 400))) {
+		nk_layout_row_dynamic(ctx, 120, 1);
+		struct nk_colorf colorf = nk_color_picker(ctx, nk_color_cf(*color), NK_RGB);
+		nk_layout_row_dynamic(ctx, 25, 1);
+		colorf.r = nk_propertyf(ctx, "#R:", 0, colorf.r, 1.0f, 0.01f,0.005f);
+		colorf.g = nk_propertyf(ctx, "#G:", 0, colorf.g, 1.0f, 0.01f,0.005f);
+		colorf.b = nk_propertyf(ctx, "#B:", 0, colorf.b, 1.0f, 0.01f,0.005f);
+
+		*color = nk_rgb_cf(colorf);
+
+		nk_combo_end(ctx);
+	}
+}
+
+// TODO style_style_item?  how to handle images if at all?
```

Comment:
Well to be useful in general I'd have to allow the user to specify the images so predefined either at compile or runtime by them and passed in somehow.  The problem is I've never even looked at an example that uses images for style items.  Do you know a good reference project/example that uses them?

Reply:
Well, there is [example/skinning.c](https://github.com/Immediate-Mode-UI/Nuklear/blob/master/example/skinning.c)

Reply:
D'oh!  I knew I should have looked before I asked.  I remember seeing that when I first learned about Nuklear but since then I've always played around with the demos and never the examples.  I'll have to spend some time going over all the examples again and maybe see what happens when I combine the configurator in it's current form with them (probably breaks completely).